docs/webiny-apps/page-builder/development/creating-iframe-element-plugin.md

```diff
@@ -0,0 +1,203 @@
+
+```jsx
+import { Tab } from "@webiny/ui/Tabs";
+import { Input } from "@webiny/ui/Input";
+import { Grid, Cell } from "@webiny/ui/Grid";
+import {
+    PbEditorPageElementAdvancedSettingsPlugin
+} from "@webiny/app-page-builder/types";
+import { validation } from "@webiny/validation";
+
+
+export default () => {
+    return {
+            name: "pb-editor-page-element-advanced-settings-iframe",
+            type: "pb-editor-page-element-advanced-settings",
+            elementType: "iframe",
+            render({ Bind }) {
+                return (
+                    <Tab icon={<IFrameIcon />} label="IFrame">
+                        <Grid>
+                            <Cell span={12}>
+                                <Bind
+                                    // Binding the iframe.url in the 'data' property
+                                    name={"iframe.url"}
+                                    validators={validation.create("required,url")}
+                                >
+                                    <Input
+                                        label={"IFrame URL"}
+                                        description={"Enter an iFrame URL"}
+                                    />
+                                </Bind>
+                            </Cell>
+                        </Grid>
+                    </Tab>
+                );
+            }
+        } as PbEditorPageElementAdvancedSettingsPlugin
+
+```
+
+This code shows the settings dialog and asks for the URL of the iframe as shown in the image below.
+
+![Settings dialog](/img/webiny-apps/page-builder/development/development/plugin-reference/editor/iframe/settings-dialog.png)
+
+> In order to add new fields to your settings dialog, you can add more `PbEditorPageElementAdvancedSettingsPlugin` repeat the same process we've just shown. This will allow you to create other plugins too.
+
+After saving the URL, the Bind component will bind the data to the state of our plugin so that the component on the render function will have access to the URL that the user provides, and it renders the IFrameEmbed component.
+
+The IFrameEmbed component:
+
+```jsx
+const IFrameEmbed = (props) => {
+  const { element } = props;
+
+  if (!element.data.iframe.url) {
+    return <div>Loading iFrame data...</div>;
+  }
+
+  return (
+    <div
+      className={
+        "webiny-pb-base-page-element-style webiny-pb-page-element-iframe " + outerWrapper
+      }
+    >
+      <div className={innerWrapper}>
+        <div id={element.id} />
+        <iframe src={element.data.iframe.url} width="100%" height={element.data.iframe.height} />
+      </div>
+    </div>
+  );
+};
+
+export default IFrameEmbed;
+```
+
+One thing to notice is that every element has it's unique CSS class such as `webiny-pb-page-element-iframe`, to add custom style to the new element plugin.
+
+And this is basically all you need to know about adding element plugins.
```

Comment:
Shouldn't we have a section in which we explain how to write a plugin that will render the plugin on an actual page?

So here we were explaining two plugins, one which will add the new page element into the editor, and the other that defines how the settings dialog will look like. That's fine, but it isn't the complete story. We still have to write a "render" plugin, which will be used on the actual website. This is mentioned in the article, but not explained.

Not sure if you've managed to see that part while you were examining this whole topic. Let me know if you'll need any help there.

Reply:
BTW, if you have the "render" plugin properly set up, you should see your page element properly rendered in here:

![image](https://user-images.githubusercontent.com/5121148/82323234-0a1ed580-99d8-11ea-94ef-32530f283a46.png)

Otherwise, you shouldn't be able to see it.